dashboard/app/controllers/admin_reports_controller.rb

```diff
@@ -88,6 +88,46 @@ def funometer_by_script_level
     end
   end
 
+  def level_answers
+    authorize! :read, :reports
+
+    @headers = ['Level ID', 'User Email', 'Data']
+    @responses = {}
+    @response_limit = 100
+    if params[:levels] || params[:mc_levels]
+      # Parse the parameters, namely the set of levels to grab answers for.
+      @mc_levels = params[:mc_levels] ? params[:mc_levels].split(',') : []
+      @levels = @mc_levels + (params[:levels] ? params[:levels].split(',') : [])
+
+      @levels.each do |level_id|
+        # Don't query for data if we've already retrieved it.
+        if @responses[level_id]
+          next
+        end
+
+        # Regardless of the level type, query the DB for level answers.
+        @responses[level_id] = LevelSource.
```

Comment:
It isn't clear to me that there is enough shared code to abstract some of it into a common function.